src/Consumers/Consumer.php

```diff
@@ -294,20 +294,27 @@ private function handleException(Throwable $exception, Message|KafkaConsumerMess
 
             report($throwable);
 
+            if (! is_null($this->config->getDlq())) {
+                $this->sendToDlq($message, $exception);
+                $this->committer->commitDlq($message);
+            }
+
             return false;
         }
     }
 
     /** Send a message to the Dead Letter Queue. */
-    private function sendToDlq(Message $message): void
+    private function sendToDlq(Message $message, Throwable $throwable): void
     {
         $topic = $this->producer->newTopic($this->config->getDlq());
         $topic->producev(
             partition: RD_KAFKA_PARTITION_UA,
             msgflags: 0,
             payload: $message->payload,
             key: $this->config->getConsumer()->producerKey($message),
-            headers: $message->headers ?? []
+            headers: $message->headers ?? [],
+            reason: $throwable->getMessage(),
```

Comment:
Hey @ebrahimradi this will not work. This is the signature of the `producev` method:

```php
public function producev($partition, $msgflags, $payload, $key = null, $headers = null, $timestamp_ms = null)
```

Reply:
this was a sudo code actually :D so do not expect if it works. we can add the reason also in header:
headers: array_merge($message->headers ?? [], ['dlq_reason' => $throwable->getMessage()])